platform/admin/core/admin-core-appservice/src/test/java/org/codice/ddf/admin/application/service/command/AbstractProfileCommandTest.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.admin.application.service.command;
+
+
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+import org.apache.karaf.bundle.core.BundleService;
+import org.apache.karaf.bundle.core.internal.BundleServiceImpl;
+import org.apache.karaf.features.FeaturesService;
+import org.apache.karaf.features.internal.service.FeaturesServiceImpl;
+import org.codice.ddf.admin.application.service.ApplicationService;
+import org.codice.ddf.admin.application.service.ApplicationServiceException;
+import org.codice.ddf.admin.application.service.impl.ApplicationServiceImpl;
+import org.junit.Before;
+import org.junit.Test;
+import org.osgi.service.resolver.ResolutionException;
+
+
+
+public class AbstractProfileCommandTest {
+
+    private ApplicationService applicationService;
+    private FeaturesService featuresService;
+    private BundleService bundleService;
+
+    private AbstractProfileCommand abstractProfileCommand;
+
+    @Before
+    public void setUp() {
+        this.applicationService = mock(ApplicationServiceImpl.class);
+        this.featuresService = mock(FeaturesServiceImpl.class);
+        this.bundleService = mock(BundleServiceImpl.class);
+
+        this.abstractProfileCommand = getCommand(applicationService, featuresService, bundleService);
+    }
+
+    @Test(expected = ApplicationServiceException.class)
+    public void testApplicationServiceFailure() throws Exception {
+        doThrow(new ApplicationServiceException()).when(applicationService).startApplication(anyString());
+        abstractProfileCommand.execute();
+    }
+
+    @Test(expected = ResolutionException.class)
+    public void testFeatureServiceFailure() throws Exception {
+        doThrow(new ResolutionException("")).when(featuresService).installFeature(anyString());
+        abstractProfileCommand.execute();
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testBundleServiceFailure() throws Exception {
+        doThrow(new IllegalArgumentException("")).when(bundleService).getBundle(anyString());
```

Comment:
Why passing in a null string? Even better, just use the class arg: `doThrow(IllegalArgumentException.class)`